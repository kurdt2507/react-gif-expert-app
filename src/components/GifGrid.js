import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const { data:imagenes, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>Loading...</p>}

            <div className="card-grid animate__animated animate__fadeIn">
                { imagenes.map((img) => (
                    <GifGridItem
                        key={ img.id }
                        { ...img }
                    />
                )) }
            </div>
        </>
    );
}
