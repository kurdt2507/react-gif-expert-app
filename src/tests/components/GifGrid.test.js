import { shallow } from 'enzyme';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre <GifGrid />', () => {
    const category = 'Caballeros del zodiaco';

    test('Valida que se cargue correctamente el componente comparando con el snapshot', () => {
        // Simula el llamado a la funci칩n useFetchGifs devolviendo los valores por defecto
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />);

        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe mostrar items cuando se cargan im치genes con useFetchGifs', () => {
        const gifs = [{
            id: "HnjlAWqut92E",
            title: "saint seiya GIF",
            url: "https://media0.giphy.com/media/HnjlAWqut92E/giphy.gif?cid=6bfbe4d0olhqc3xwt6ktg3gopquq1njjdj02f07tkbrsc99t&rid=giphy.gif"
        }];
        // Simula el llamado a la funci칩n useFetchGifs devolviendo el arreglo de im치genes
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category } />);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
