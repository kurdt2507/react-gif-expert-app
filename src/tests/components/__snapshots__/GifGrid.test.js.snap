// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGrid /> Valida que se cargue correctamente el componente comparando con el snapshot 1`] = `
<Fragment>
  <h3>
    Caballeros del zodiaco
  </h3>
  <p
    className="animate__animated animate__pulse"
  >
    Loading...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
